code/modules/supply/supply_packs/pack_shuttle.dm

```diff
@@ -0,0 +1,109 @@
+	// these are special, but start enabled. As soon as one is ordered, all variants of this should be disabled.
+	special = TRUE
+	special_enabled = TRUE
+	/// The shuttle that this pack corresponds to ordering.
+	/// If given on init, will fill out the pack automagically.
+	var/datum/map_template/shuttle/template
+	/// The base speed multiplier to apply to the shuttle. 2 means it arrives twice as fast.
+	var/speed_factor = 1
+
+
+/datum/supply_packs/abstract/shuttle/New(datum/map_template/shuttle/_template)
+	. = ..()
+	if(_template)
+		template = _template
+	if(template)
+		if(ispath(template))
+			template = new template()
+		name = template.name
+		manifest = template.description
+		ui_manifest = list(manifest)
+
+/datum/supply_packs/abstract/shuttle/Destroy(force, ...)
+	. = ..()
+	qdel(template)
+
+
+/datum/supply_packs/abstract/shuttle/can_order()
+	// check here so people don't spend their money on it
+	return (SSshuttle.emergency?.mode == SHUTTLE_IDLE)
+
+/datum/supply_packs/abstract/shuttle/on_order_confirm(datum/supply_order/order)
+	. = ..()
+	if(SSshuttle.custom_shuttle_ordered)
+		return
+	if(isnull(template))
+		CRASH("Shuttle pack [src] has no map template to load!")
+	GLOB.major_announcement.Announce("[order.orderedby] has purchased [name] for [cost] credit\s as the emergency shuttle for the shift.", "Shuttle Purchase Receipt")
+	var/orderer = order.orderedby
+	log_and_message_admins("Shuttle pack [src] costing [cost] has been ordered and paid for by [orderer] ([order.orderedbyRank])")
+	for(var/pack_key in SSeconomy.supply_packs)
+		var/datum/supply_packs/abstract/shuttle/SP = SSeconomy.supply_packs[pack_key]
+		if(!istype(SP))
+			continue
+		SP.special_enabled = FALSE
+	SSshuttle.custom_shuttle_ordered = TRUE
+	var/obj/docking_port/mobile/shuttle = SSshuttle.load_template(template)
+	shuttle.shuttle_speed_factor = speed_factor
+	SSshuttle.replace_shuttle(shuttle)
+
+/datum/supply_packs/abstract/admin_notify/donations
+	name = "Donation to Lonely Corgi Foundation"
+	manifest = "1% of every donation goes towards supporting corgis in need."
+	cost = 500
+
+// these ones are pretty reasonable
+
+/datum/supply_packs/abstract/shuttle/bar
+	cost = 3250
+	template = /datum/map_template/shuttle/emergency/bar
+
+/datum/supply_packs/abstract/shuttle/dept
+	cost = 3000
+	template = /datum/map_template/shuttle/emergency/dept
+
+/datum/supply_packs/abstract/shuttle/old
+	cost = 2500
+	template = /datum/map_template/shuttle/emergency/old
+
+// this one isn't great but it isn't horrible either
+
+/datum/supply_packs/abstract/shuttle/cramped
+	cost = 3750
+	template = /datum/map_template/shuttle/emergency/cramped
+
+
+// these, otoh, have some pretty silly features, and are hidden behind emag
+
+/datum/supply_packs/abstract/shuttle/military
+	// if you can pool enough money together for this one, maybe you deserve it
+	speed_factor = 1.5
+	cost = 4250
+	template = /datum/map_template/shuttle/emergency/military
```

Comment:
Shouldn't this have the emag hidden variable? Or is the comment incorrect?

Reply:
I kind of changed my mind for that one, in favor of making it very expensive